src/Chapter2.hs

```diff
@@ -621,7 +634,12 @@ Write a function that takes elements of a list only in even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [a] -> [a]
+takeEven [] = []
+takeEven [x] = [x]
+takeEven [x, _] = [x]
```

Comment:
That's right! 💪🏼 

If you put these two cases after the last check here, you could join them into just one case:

```haskell
takeEven (x:_) = [x]
```

Reply:
Oh, I see, just got used to having the most complicated logic at the end. I think the `takeEven [x, _] = [x]` case is actually redundant. How about just removing it:

```
takeEven :: [a] -> [a]
takeEven [] = []
takeEven [x] = [x]
takeEven (x : _ : xs) = x : takeEven xs
```

Because `xs` in the last case can be an empty list so any list with just 2 elements is also covered with that case.

Reply:
Yes, this is absolutely right and quite elegant 💯